app/user/page.js

```diff
@@ -1,17 +1,259 @@
+import { Calendar, momentLocalizer, Views } from "react-big-calendar";
+import moment from "moment";
+import { useRouter } from 'next/navigation';
+import "react-big-calendar/lib/css/react-big-calendar.css";
+import StyledPaper from "../components/styledComponents/StyledPaper";
+import { BlueButton } from "../components/styledComponents/StyledButton";
+import DesignTitel from "../components/styledComponents/DesignTitel";
 
-import React, { useState } from 'react';
-import { Box } from '@mui/material';
+const Dashboard = () => {
+  const localizer = momentLocalizer(moment);
+  const router = useRouter();
 
-function UserDashboard() {
+  const handleCreateEvent = () => {
+    router.push('/user/createEvent'); // Navigate to /user/createEvent
+  };
+
+  // Initiale Events
+  const [events, setEvents] = useState([
+    {
+      id: 1,
+      title: "Belegte Veranstaltung",
+      start: new Date(2024, 10, 29, 10, 0),
+      end: new Date(2024, 10, 29, 12, 0),
+      type: "booked",
+      description: "Dies ist eine belegte Veranstaltung.",
+    },
+    {
+      id: 2,
+      title: "Zeitnah anstehende Veranstaltung",
+      start: new Date(2024, 10, 28, 14, 0),
+      end: new Date(2024, 10, 28, 16, 0),
+      type: "upcoming",
+      description: "Diese Veranstaltung findet bald statt.",
+    },
+  ]);
+
+  // State-Management
+  const [anchorEl, setAnchorEl] = useState(null);
+  const [selectedEventId, setSelectedEventId] = useState(null);
+  const [newEventModalOpen, setNewEventModalOpen] = useState(false);
+  const [newEvent, setNewEvent] = useState({
+    title: "",
+    description: "",
+    start: null,
+    end: null,
+  });
+
+  // Optionen-Menü
+  const handleMenuOpen = (event, eventId) => {
+    setAnchorEl(event.currentTarget);
+    setSelectedEventId(eventId);
+  };
+
+  const handleMenuClose = () => {
+    setAnchorEl(null);
+    setSelectedEventId(null);
+  };
+
+  const handleOptionClick = (option) => {
+    console.log(`Option "${option}" für Event ID ${selectedEventId} gewählt`);
+    handleMenuClose();
+  };
+
+  // Hinzufügen eines neuen Termins
+  const handleSelectSlot = (slotInfo) => {
+    setNewEvent({
+      ...newEvent,
+      start: slotInfo.start,
+      end: moment(slotInfo.start).add(1, "hours").toDate(),
+    });
+    setNewEventModalOpen(true);
+  };
+
+  const handleNewEventChange = (field, value) => {
+    setNewEvent((prev) => ({ ...prev, [field]: value }));
+  };
+
+  const handleAddNewEvent = () => {
+    if (newEvent.title && newEvent.start && newEvent.end) {
+      setEvents((prevEvents) => [
+        ...prevEvents,
+        { id: prevEvents.length + 1, ...newEvent, type: "myEvent" },
+      ]);
+      setNewEventModalOpen(false);
+      setNewEvent({
+        title: "",
+        description: "",
+        start: null,
+        end: null,
+      });
+    }
+  };
+
+  // Event-Styling
+  const eventStyleGetter = (event) => {
```

Comment:
Hier wäre eine Legende am Kalender vielleicht gut.

Reply:
wie meinst du es genau mit legende.

Reply:
Was welche Farbe bedeutet. Vllt mit nem Popover oder so

Reply:
Jetzt ist die Farbe raus, die ist aber in #48 noch drin. Meiner Meinung nach sind das 2 Seiten/features die in getrennten Branches sein sollten. Also entsprechend /user aus #48 entfernen, aber gerne die Änderungen auf die Version hier anpassen